java/compaction/compaction-job-creation/src/test/java/sleeper/compaction/job/creation/CreateCompactionJobsIT.java

```diff
@@ -112,6 +122,35 @@ public void shouldCompactAllFilesInSinglePartition() throws Exception {
                 });
     }
 
+    @Test
+    public void shouldSendAssignJobIdRequestToSqs() throws Exception {
+        // Given
+        tableProperties.set(COMPACTION_JOB_ID_ASSIGNMENT_COMMIT_ASYNC, "true");
+        FileReferenceFactory fileReferenceFactory = FileReferenceFactory.from(stateStore);
+        FileReference fileReference1 = fileReferenceFactory.rootFile("file1", 200L);
+        FileReference fileReference2 = fileReferenceFactory.rootFile("file2", 200L);
+        FileReference fileReference3 = fileReferenceFactory.rootFile("file3", 200L);
+        FileReference fileReference4 = fileReferenceFactory.rootFile("file4", 200L);
+        List<FileReference> files = List.of(fileReference1, fileReference2, fileReference3, fileReference4);
+        stateStore.addFiles(files);
+
+        // When
+        jobCreator().createJobs(tableProperties);
+
+        // Then
+        assertThat(stateStore.getFileReferencesWithNoJobId())
+                .usingRecursiveFieldByFieldElementComparatorIgnoringFields("lastStateStoreUpdateTime")
+                .containsExactlyInAnyOrderElementsOf(files);
+        assertThat(receiveJobQueueMessage().getMessages())
+                .extracting(this::readJobMessage).singleElement().satisfies(job -> {
+                    assertThat(job.getInputFiles()).containsExactlyInAnyOrder("file1", "file2", "file3", "file4");
+                    assertThat(job.getPartitionId()).isEqualTo("root");
+                    assertThat(receiveAssignmJobIdQueueMessage().getMessages())
```

Comment:
It might be slightly easier to read if we receive both the job and file assignment messages before we start asserting. It took me a minute to figure out there was only one job, it looked like we were receiving messages from the queue more than once.